### DISCUSSION TITLE
Using only 1 CPU core

### DISCUSSION THREAD

**Discussion by hadarshavit** [intel/scikit-learn-intelex#1995] [Q&A]
Hi, I'm trying to make my experiment as reproducible as possible (including running time).
Therefore, I'm want to limit the CPU usage to one core.
I used OMP_NUM_THREADS=1, OPENBLAS_NUM_THREADS=1,  MKL_NUM_THREADS=1, BLIS_NUM_THREADS=1.
This made "normal" sklearn models to use only one CPU core, but for models using the intel extension (especially sklearn) I see a CPU usage of more than 100% in htop.
How can I fix that?
Thanks

**Comment by Alexsandruss** [comment]
`OMP_NUM_THREADS`, `OPENBLAS_NUM_THREADS` and `MKL_NUM_THREADS` should work if associated libraries are installed.

Running script with enabled sklearnex debugging might show if there is a problem:
```bash
OMP_NUM_THREADS=1 OPENBLAS_NUM_THREADS=1 MKL_NUM_THREADS=1 BLIS_NUM_THREADS=1 SKLEARNEX_VERBOSE=DEBUG python test_svr.py
```

My output in this case is correct - SVR is patched and number of threads is reduced to 1:
```
DEBUG:sklearnex: sklearnex.svm.svr.SVR.fit: setting 1 threads (previous - 8)
INFO:sklearnex: sklearn.svm.SVR.fit: running accelerated version on CPU
INFO:sklearnex: sklearn.utils.validation._assert_all_finite: running accelerated version on CPU
```
